portal-impl/src/com/liferay/portal/security/auth/AuthVerifierPipeline.java

```diff
@@ -0,0 +1,477 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.security.auth;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.auth.AccessControlContext;
+import com.liferay.portal.kernel.security.auth.verifier.AuthVerifier;
+import com.liferay.portal.kernel.security.auth.verifier.AuthVerifierConfiguration;
+import com.liferay.portal.kernel.security.auth.verifier.AuthVerifierResult;
+import com.liferay.portal.kernel.service.UserLocalServiceUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.registry.Filter;
+import com.liferay.registry.Registry;
+import com.liferay.registry.RegistryUtil;
+import com.liferay.registry.ServiceReference;
+import com.liferay.registry.ServiceTracker;
+import com.liferay.registry.ServiceTrackerCustomizer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.servlet.http.HttpServletRequest;
+
+import jodd.util.Wildcard;
+
+/**
+ * @author Tomas Polesovsky
+ * @author Peter Fellwock
+ * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+ */
+@Deprecated
+public class AuthVerifierPipeline {
+
+	public static final String AUTH_TYPE = "auth.type";
+
+	public static String getAuthVerifierPropertyName(String className) {
+		String simpleClassName = StringUtil.extractLast(
+			className, StringPool.PERIOD);
+
+		return StringBundler.concat(
+			PropsKeys.AUTH_VERIFIER, simpleClassName, StringPool.PERIOD);
+	}
```

Comment:
hey @arthurchan35, 

I guess we need to move this method elsewhere? as I can see it is still used but maybe we don't even need to have this method

Reply:
Hey @csierra ,
Yes we can move it to some where else, but I think we still need to leave it in this deprecated pipeline class for at least one portal version, as theoretically custom code can use it.

Maybe we can do the move/remove this method in the next version, when we are going to remove this deprecated class.

Reply:
hey, 
so I guess then we should provide a replacement as of today so people can move away from the deprecated class to the new method, unless we plan to remove the need for the method completely.